src/analysisd/analysisd.c

```diff
@@ -1661,8 +1661,11 @@ RuleInfo *OS_CheckIfRuleMatch(Eventinfo *lf, RuleNode *curr_node)
     /* If it is a context rule, search for it */
     if(currently_rule->context == 1)
     {
-        if(!currently_rule->event_search(lf, currently_rule))
-            return(NULL);
+
+	if(!(currently_rule->context_opts & SAME_DODIFF)) {
+        	if(!currently_rule->event_search(lf, currently_rule))
+            	return(NULL);
```

Comment:
This is personal choice I hate bare if blocks as they get hard to read.  I know it more typing but what i did here is always more clear to me when reading: 

```
if(!(currently_rule->context_opts & SAME_DODIFF)) {
    if(!currently_rule->event_search(lf, currently_rule)){
        return(NULL)
    }           
}
```